### DISCUSSION TITLE
Ensure records immutability

### DISCUSSION THREAD

**Discussion by D2O84** [dotnet/csharplang#5333] [Language Ideas] [answered]
Would it be possible to add an option to ensure records immutability, such as records would only allow reference to primitive types, records, ReadOnlyCollection<Record types or primitive or ...>, ReadOnlyDictionary< ... >, ...?

in csproj
```
<PropertyGroup>
    ...
    <ImmutableRecord>true</ImmutableRecord>
</PropertyGroup>
```

Thank you

**Comment by HaloFour** [comment, marked as answer]
You can use an analyzer in your project to enforce that records are effectively immutable: https://docs.microsoft.com/en-us/visualstudio/extensibility/getting-started-with-roslyn-analyzers?view=vs-2019
